src/dash/DefaultDash.jsx

```diff
@@ -86,20 +86,77 @@ const tutors = [
         rating: 1,
         online:true,
         gender: 'male'
+    },
+    {
+        first_name: 'John',
+        last_name: 'Doe',
+        subjects: [{subject:'Architecture', start_grade:7, end_grade:11},{subject:'Computer Science', start_grade:8, end_grade:9},{subject:'Biology', start_grade:8, end_grade:12}],
+        availability: {
+            Monday: [],
+            Tuesday: [],
+            Wednesday: [],
+            Thursday: [],
+            Friday: [{start_time:'12:00', end_time:'1:00'},{start_time:'2:00', end_time:'3:00'},{start_time:'4:00', end_time:'5:00'}],
+            Saturday: [],
+            Sunday: []
+        },
+        profile_picture: '/images/default_user_img.png',
+        bio: 'Hello, my name is John. Please ask me for help, I am smart.',
+        email: 'jdoe6@gatech.edu',
+        class_standing: 'Freshman',
+        rating: 2.5,
+        online:false,
+        gender: 'male'
     }
 ];
 
 class DefaultDash extends React.Component {
     constructor(props) {
         super(props);
+        this.state = {
+            searchType:"online",
+            displayTutors: this.getDisplayTutors("online"),
+            searchTime:"ASAP",
+            searchSubject:null
+        }
+        this.handleSearchClicked = this.handleSearchClicked.bind(this);
+        this.getDisplayTutors = this.getDisplayTutors.bind(this);
+
+    }
+
+
+    handleSearchClicked(subject, time) {
+        this.setState({searchType:"searchResults"});
+        this.setState({displayTutors: this.getDisplayTutors("searchResults", subject, time)});
+        this.setState({searchSubject:subject});
+        this.setState({searchTime:time});
+        this.forceUpdate();
+    }
+
+    getDisplayTutors(searchType, subject, time) {
+        var newTutorsList;
+        if (searchType === "online") {
+            newTutorsList = tutors.filter((obj, num) => {
+                return obj.online;
+            });
+        } else {
+            newTutorsList = tutors.filter((obj, num) => {
```

Comment:
Code in the backend `user_dao` that does this (not merged into dev yet, in the branch `tutor-filtering`)